CHANGELOG.md

```diff
@@ -4,6 +4,23 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 ## Unreleased
 
+- The following components now take an optional `loadingComponent` to be rendered while fetching data:
+  - `Value`
+  - `Text`
+  - `Link`
+  - `Image`
+  - `Video`
+- When these are not passed, the components render a default message instead. 
+
+- The following components now take an optional `errorComponent` to be rendered in case of error:
+  - `Value`
+  - `Text`
+  - `Link`
+- When these are not passed, the components render a default message instead. 
+
+- `loading` changed to `loadingComponent` in `DatasetProvider` to match the rest of the components
```

Comment:
As it is written right now this is a breaking change and would require a major version bump.

Instead, I'd mark `loading` as deprecated (and replaced by `loadingComponent`) instead, and have `DatasetProvider` look for `loading` if `loadingComponent` is not provided.